Source/AGXUnreal/Public/Terrain/AGX_MovableTerrainComponent.h

```diff
@@ -0,0 +1,424 @@
+#pragma once
+
+// AGX Dynamics for Unreal include.s
+#include "AGX_NativeOwner.h"
+#include "Terrain/TerrainBarrier.h"
+#include "Terrain/AGX_TerrainMeshUtilities.h"
+#include "AGX_ShovelReference.h"
+
+#include "CoreMinimal.h"
+#include "ProceduralMeshComponent.h"
+#include "AGX_MovableTerrainComponent.generated.h"
+
+
+class UAGX_ShapeComponent;
+class UAGX_TerrainMaterial;
+class UAGX_ShovelComponent;
+class UNiagaraSystem;
+class UNiagaraComponent;
+
+
+
+
+UENUM(BlueprintType, Category = "AGX Terrain Mesh")
+enum class EAGX_MeshTilingPattern : uint8
+{
+	None UMETA(DisplayName = "None"),
+	StretchedTiles UMETA(DisplayName = "Stretched Tiles")
+};
+
+struct MeshTile
+{
+	int MeshIndex;
+	FIntVector2 Resolution;
+	FVector Center;
+	FVector2D Size;
+	FVector2D UvScale;
+	FBox2D BoundingBox;
+	bool IsSkirt;
+	bool IsReverseWinding;
+
+	MeshTile(
+		int MeshSectionIndex, FVector TileCenter, FVector2D TileSize, FIntVector2 TileRes,
+		FVector2D TileUvScale, bool IsTileSkirt, bool IsTileReverseWinding)
+	{
+		MeshIndex = MeshSectionIndex;
+		Center = TileCenter;
+		Size = TileSize;
+		Resolution = TileRes;
+		UvScale = TileUvScale;
+		IsSkirt = IsTileSkirt;
+		IsReverseWinding = IsTileReverseWinding;
+	}
+};
+
+using TiledMesh = TArray<MeshTile>;
+
+UENUM(BlueprintType, Category = "AGX Terrain Mesh")
+enum class EAGX_MeshType : uint8
+{
+	None UMETA(DisplayName = "None"),
+	Terrain UMETA(DisplayName = "Terrain"),
+	DebugPlane UMETA(DisplayName = "DebugPlane"),
+	Collision UMETA(DisplayName = "Collision"),
+	BackBed UMETA(DisplayName = "BackBed"),
+};
+/**
+ *
+ */
+UCLASS(ClassGroup = "AGX_Terrain", Category = "AGX", Meta = (BlueprintSpawnableComponent),
+	Hidecategories = (Cooking, Collision, Input, LOD, Physics, Replication, Materials, ProceduralMesh))
+class AGXUNREAL_API UAGX_MovableTerrainComponent : public UProceduralMeshComponent,
+												   public IAGX_NativeOwner
+{
+	GENERATED_BODY()
+public:
+	UAGX_MovableTerrainComponent(const FObjectInitializer& ObjectInitializer);
+
+	virtual void BeginPlay() override;
+	virtual void EndPlay(const EEndPlayReason::Type EndPlayReason) override;
+
+	virtual void TickComponent(
+		float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction);
+
+
+	virtual void PostInitProperties() override;
+	virtual void PostLoad() override; // When loaded in Editor or Game
+#if WITH_EDITOR
+	virtual void PostEditChangeChainProperty(FPropertyChangedChainEvent& Event) override;
+#endif
+	UPROPERTY(EditAnywhere, Category = "AGX Editor")
+	bool bRebuildMesh = false;
+	UPROPERTY(EditAnywhere, Category = "AGX Editor")
+	bool bShowDebugPlane = false;
+	UPROPERTY(EditAnywhere, Category = "AGX Editor")
+	bool bShowUnrealCollision = false;
+	UPROPERTY(EditAnywhere, Category = "AGX Editor")
+	bool bHideTerrain = false;
```

Comment:
I think this one is covered by the `bVisible` member of the UProceduralMeshComponent. 

![image](https://github.com/user-attachments/assets/b2ae32c8-df2d-4dca-ba87-767a729b2b65)